### PR TITLE
Motion Primitives Readme

### PR DESCRIPTION
**Pull request by orduno**
Updated the `nav2_motion_primitives` documentation.

### CODE REVIEW

File: nav2_motion_primitives/README.md [hunk 1]
```diff
@@ -1,2 +1,115 @@
...
+```
+
+### (optional) Define the Behavior Tree action node
+
+For using motion primitivies within a behavior tree such as [bt_navigator](../nav2_bt_navigator/README.md##Navigation-Behavior-Trees), then a corresponding *action node* needs to be defined. Checkout `nav2_tasks` for examples on how to implement one.
+
+## Plans
+
+- Check for collision before executing a primitive. Issues [379](https://github.com/ros-planning/navigation2/issues/379) and [533](https://github.com/ros-planning/navigation2/issues/533).
+- Remove the stop primitive, move the funcionality to the robot class. Issue [575](https://github.com/ros-planning/navigation2/issues/575)
```
Comment (crdelsey) [inline nav2_motion_primitives/README.md]: The robot class could be used to send a cmd_vel of 0, but if we want the robot to come to a controlled stop that obeys the kinematic limits of the robot, then a stop primitive is still appropriate, IMHO.

File: nav2_motion_primitives/README.md [hunk 2]
```diff
@@ -1,2 +1,115 @@
...
+- Consider moving `nav2_motion_primitives` altogether to the `nav2_robot` package. Issue [378](https://github.com/ros-planning/navigation2/issues/378).
+- Define primitives that describe arcs.
```
Comment (crdelsey) [inline nav2_motion_primitives/README.md]: Do you really think these extra primitives and primitive features are worth it? Everything the motion primitives do can also be done by using the rest of the nav stack.

**Comment by crdelsey** [comment]
Oh. One other thought. I think we should include a warning that this is still in development and only works with Turtlebot 3 at the moment. I think a higher priority than adding more motion primitives would be to make the existing motion primitives configurable for other robots.

File: nav2_motion_primitives/README.md [hunk 3]
```diff
@@ -1,2 +1,115 @@
-# Behaviors
-TODO (orduno)
+# Motion Primitives
+
+The `nav2_motion_primitives` package implements, as the name suggests, a module for executing simple controlled <!--and collision-free--> robot movements such as rotating on its own axis or moving linearly.
```
[the remark this answers is not in the document]
Comment (orduno) [reply nav2_motion_primitives/README.md]: It's commented out, a reminder to add something about free space checking, but sure, I'll remove it.

File: nav2_motion_primitives/README.md [hunk 4]
```diff
@@ -1,2 +1,115 @@
...
+- Possibly do some initialization.
+- Return a `nav2_tasks::TaskStatus` given the initial checks.
+
+The `onCycleUpdate` method is called periodically until it returns `FAILED` or `SUCCEEDED`, here we should:
+- Set the robot in motion.
+- Perform some unit of work.
+- Check if the robot state, determine if work completed
+- Return a `nav2_tasks::TaskStatus`.
+
+### Executing a primitive
```
[the remark this answers is not in the document]
Comment (orduno) [reply nav2_motion_primitives/README.md]: Sorry, the title was wrong. Fixed it.

File: nav2_motion_primitives/README.md (hunk 1 above)
Comment (orduno) [reply nav2_motion_primitives/README.md]: Yes, we can have different types of stop. In this case, we just send a velocity command of zero and it's not really a primitive. A stop primitive would, as you mention, look into the robot dynamics and world restrictions.

File: nav2_motion_primitives/README.md (hunk 2 above)
Comment (orduno) [reply nav2_motion_primitives/README.md]: Or navigation could be done entirely by linking primitives :) 

Depending on the feedback from the community we might want to develop this package and support a wide variety of primitives to support all kinds of task, navigation, docking, etc.

**Comment by orduno** [comment]
Thanks, I've addressed the feedback.

**Pull request merged**